File: backend/src/main/java/com/elmenture/core/payload/SignInResponse.java
package com.elmenture.core.payload;

import lombok.Data;

import java.util.List;

@Data
public class SignInResponse {
    private boolean success = false;
    private boolean isNewAccount = false;
    private String sessionKey;
    private String email;
    private UserMeasurementsDto actualMeasurements;
    private FemaleSize femaleSize;
    private String s3AccessKeyId;
    private String s3SecretKeyId;
    private String contactPhoneNumber;
    private String mobileMoneyNumber;
    private String physicalAddress;
    private String clothingRecommendations;
    private String name;
    private boolean staff;
    private List<TagPropertyDto> tagProperties;


    /**
     * Sizes that have been translated from what the user entered
     */
    @Data
    public static class FemaleSize {
        SizeDto dress;
    }
}

File: android/common/src/main/java/models/SignInResponse.kt
package models

import com.google.gson.annotations.SerializedName

class SignInResponse {
    @field:SerializedName("success")
    var isSuccessful: Boolean = false

    @field:SerializedName("isNewAccount")
    var isNewAccount: Boolean = false

    @field:SerializedName("sessionKey")
    var sessionKey: String? = null

    @field:SerializedName("email")
    var email: String? = null

    @field:SerializedName("staff")
    var isStaff: Boolean = false

    @field:SerializedName("s3AccessKeyId")
    var s3AccessKeyId: String? = null

    @field:SerializedName("s3SecretKeyId")
    var s3SecretKeyId: String? = null

    @field:SerializedName("actualMeasurements")
    var actualMeasurements: ActualMeasurements? = null

    @field:SerializedName("femaleSize")
    var femaleSize: FemaleSize? = null

    @field:SerializedName("contactPhoneNumber")
    var contactPhoneNumber: String? = null

    @field:SerializedName("clothingRecommendations")
    var clothingRecommendations: String? = null

    @field:SerializedName("physicalAddress")
    var physicalAddress: String? = null

    @field:SerializedName("name")
    var name: String? = null
}